addons/hr_attendance/wizard/hr_attendance_error.py

```diff
@@ -1,48 +1,31 @@
 
 
-class hr_attendance_error(osv.osv_memory):
+class HrAttendanceError(models.TransientModel):
 
     _name = 'hr.attendance.error'
     _description = 'Print Error Attendance Report'
-    _columns = {
-        'init_date': fields.date('Starting Date', required=True),
-        'end_date': fields.date('Ending Date', required=True),
-        'max_delay': fields.integer('Maximum Tolerance (in minutes)', required=True,
-            help="Allowed difference in minutes between the signin/signout and the timesheet computation for one sheet. Set this to 0 for no tolerance.")
-    }
-    _defaults = {
-         'init_date': lambda *a: time.strftime('%Y-%m-%d'),
-         'end_date': lambda *a: time.strftime('%Y-%m-%d'),
-         'max_delay': 120,
-    }
 
-    def print_report(self, cr, uid, ids, context=None):
-        emp_ids = []
-        data_error = self.read(cr, uid, ids, context=context)[0]
-        date_from = data_error['init_date']
-        date_to = data_error['end_date']
-        cr.execute("SELECT id FROM hr_attendance WHERE employee_id IN %s AND to_char(name,'YYYY-mm-dd')<=%s AND to_char(name,'YYYY-mm-dd')>=%s AND action IN %s ORDER BY name" ,(tuple(context['active_ids']), date_to, date_from, tuple(['sign_in','sign_out'])))
-        attendance_ids = [x[0] for x in cr.fetchall()]
+    init_date = fields.Date('Starting Date', required=True, default=lambda *a: fields.Date.today())
+    end_date = fields.Date('Ending Date', required=True, default=lambda *a: fields.Date.today())
+    max_delay = fields.Integer('Maximum Tolerance (in minutes)', required=True, default=120,
+                help="Allowed difference in minutes between the signin/signout and the timesheet computation for one sheet. Set this to 0 for no tolerance.")
+
+    @api.multi
+    def print_report(self):
+        emp_ids = set()
+        date_from = self.init_date
+        date_to = self.end_date
+        self.env.cr.execute("SELECT id FROM hr_attendance WHERE employee_id IN %s AND to_char(name, 'YYYY-mm-dd')<=%s AND to_char(name, 'YYYY-mm-dd')>=%s AND action IN %s ORDER BY name", (tuple(self.env.context['active_ids']), date_to, date_from, tuple(['sign_in', 'sign_out'])))
+        attendance_ids = [x[0] for x in self.env.cr.fetchall()]
         if not attendance_ids:
             raise UserError(_('No records are found for your selection!'))
-        attendance_records = self.pool.get('hr.attendance').browse(cr, uid, attendance_ids, context=context)
-
-        for rec in attendance_records:
-            if rec.employee_id.id not in emp_ids:
-                emp_ids.append(rec.employee_id.id)
-        data_error['emp_ids'] = emp_ids
-        datas = {
-             'ids': [],
-             'model': 'hr.employee',
-             'form': data_error
-        }
-        return self.pool['report'].get_action(
-            cr, uid, [], 'hr_attendance.report_attendanceerrors', data=datas, context=context
-        )
+        for rec in self.env['hr.attendance'].browse(attendance_ids):
```

Comment:
This browse will be removed if we use search instead of sql injection at line 23